client/src/main/java/io/pravega/client/stream/impl/PositionInternal.java

```diff
@@ -40,7 +40,7 @@
      *
      * @return the read offset for each segment in the ownedSegments set
      */
-    abstract Map<Segment, Long> getOwnedSegmentsWithOffsets();
+    public abstract Map<Segment, Long> getOwnedSegmentsWithOffsets();
```

Comment:
This was something we very deliberately avoided in the way we layed out the directory structure. This object is returned to users, so even though it is inside of the impl directory there is a temptation to call methods on it. Let's try to work out a way to avoid making this publicly callable. Perhaps by having the map passed to the caller rather than the position object?

Reply:
Looks like this is only invoked from `ModelHelper.getSegmentsFromPositions` which is actually not used anywhere. Same with the one below (invoked in `ModelHelper.toSegmentOffsetMap` which is not used anywhere.

I deleted those methods in `ModelHelper` and reverted the changes in `PositionInternal.java`.